### ISSUE TITLE
Can't commit with magit, seeing errors related to `/var/folders`

### ISSUE THREAD

**Issue by ezmiller** [syl20bnr/spacemacs#12534]
#### Description :octocat:
I recently had to migrate to a new mac using the mac migration tool. After doing so,
I now am getting access errors related to a directory `/var/folders/8s`. On my new mac,
this folder `8s` does not exist.

One error that pops up, appears on startup:

> Error running timer ‘mac-setup-help-topics’: (file-missing "Creating file with prefix" "No such file or directory" "/var/folders/8s/y0zrtc6n38j0gwnvz7_l4q1h0000gn/T/jka-com")

The other, much more consequential, occurs when I try to start a commit using magit. When I hit
`c-c` from the magit status screen, I get a message:

> files--ensure-directory: Creating directory: Permission denied, /var/folders/8s

To fix this so far, I've tried:

1) Removing the `.emacs.d` folder and then reinstalling spacemacs from the repo.
2) I'm using emacs for mac, and I tried to reinstall that.

Does anyone know how to fix this? It seems as though it should be possible to get emacs to regenerate whatever it's keeping in these folders...

#### Reproduction guide :beetle:
- I'm not sure...

*Observed behaviour:* :eyes: :broken_heart:
Errors, plus I'm unable to commit with magit.

*Expected behaviour:* :heart: :smile:
I shouldn't see errors, and I shoudl be able to commit with magit.

#### System Info :computer:
- OS: darwin
- Emacs: 26.2
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. e9afae228)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(csv ruby sql yaml typescript coffeescript markdown clojure
     (python :variables python-test-runner
             '(pytest nose)
             python-sort-imports-on-save t python-enable-yapf-format-on-save t)
     php html terraform
     (helm :variables helm-use-fuzzy 'source)
     auto-completion emacs-lisp git multiple-cursors
     (neotree :variables neo-window-fixed-size nil)
     react prettier org
     (shell :variables shell-default-shell 'multi-term shell-default-width '45 shell-default-position 'right)
     (version-control :variables version-control-diff-tool 'git-gutter version-control-diff-side 'left))
```
- System configuration features: NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS THREADS LCMS2

**Comment by duianto** [comment]
Here are a couple of pages that mention the beginning of the error message:
>Error running timer ‘mac-setup-help-topics’:

- https://www.reddit.com/r/emacs/comments/3dirjv/empty_path_in_emacsmacappspacemacs_mac_os_x/
- I installed homebrew emacs port and got following failure. https://github.com/railwaycat/homebrew-emacsmacport/issues/167
- https://github.com/hlissner/doom-emacs/issues/1322#issuecomment-483135501

A suggestion on the first page mentions adding the `osx` layer. The comment was from 4 years ago, and the author of that page said that it didn't help, but the full error message isn't the same as yours.

It's at least something to try.

It might not be a Spacemacs specific issue, because at least the last two pages don't mention Spacemacs.

**Comment by tko** [comment]
The `/var/folders/<gibberish>` is probably `TMPDIR` on macos. It changes and the related folder is created on every boot, probably every login. I think the default setup for `.spacemacs.env` included TMPDIR in to; so on the first run it works, but after reboot/relogin TMPDIR points to a non-existent directory and all kinds of trouble starts to happen.

`.spacemacs.env` should most probably not be setting `TMPDIR` or of it does, then someone needs to make sure that folder exists and is writable etc.

**Comment by gcentauri** [comment]
Ah, @tko is right. I just looked on my mac setup and there is a `TMPDIR` like `/var/folders/n8/...`  I can't seem to find anything in spacemacs explicitly setting it.

**Comment by duianto** [comment]
`.spacemacs.env` can be opened with `SPC f e e`.
If it's renamed (as a backup) or deleted, then it can be recreated with `SPC f e E` or by restarting Spacemacs.

**Comment by ezmiller** [comment]
Thank you @tko @gcentauri. Renaming `.spacemacs.env`, and then restarting emacs, solved it for me.

**Comment by ezmiller** [comment]
Should this issue be closed, or there a broader change that we should consider here?

**Comment by duianto** [comment]
There is a `.speacemacs.env` related PR:
[core] Environment variable caching fixes #12417

but I don't know if it's related or has an effect on this issue.
